File: dotnet-clients/web-client/Models/ConversionViewModel.cs
using System.ComponentModel.DataAnnotations;
using Microsoft.AspNetCore.Mvc.Rendering;
using System.Collections.Generic;

namespace web_client.Models
{
    public class ConversionViewModel
    {
        [Required(ErrorMessage = "Debe seleccionar una categoria.")]
        [Display(Name = "Categoria")]
        public string Category { get; set; } = "Mass";

        [Required(ErrorMessage = "Debe ingresar un valor a convertir.")]
        [Range(0.0001, double.MaxValue, ErrorMessage = "El valor debe ser mayor a 0.")]
        [Display(Name = "Valor a Convertir")]
        public double? InputValue { get; set; }

        [Required(ErrorMessage = "Debe seleccionar la unidad de origen.")]
        [Display(Name = "Desde")]
        public string FromUnit { get; set; }

        [Required(ErrorMessage = "Debe seleccionar la unidad de destino.")]
        [Display(Name = "Hacia")]
        public string ToUnit { get; set; }

        public double? ResultValue { get; set; }
        public string? ErrorMessage { get; set; }
        public string? SuccessMessage { get; set; }

        public List<SelectListItem> Categories { get; } = new List<SelectListItem>
        {
            new SelectListItem { Value = "Mass", Text = "Masa" },
            new SelectListItem { Value = "Length", Text = "Longitud" },
            new SelectListItem { Value = "Temperature", Text = "Temperatura" }
        };

        public static Dictionary<string, string[]> UnitsByCategory { get; } = new Dictionary<string, string[]>
        {
            { "Mass", new[] { "KILOGRAM", "GRAM", "POUND", "OUNCE" } },
            { "Length", new[] { "METER", "KILOMETER", "CENTIMETER", "MILE", "YARD", "FOOT", "INCH" } },
            { "Temperature", new[] { "CELSIUS", "FAHRENHEIT", "KELVIN" } }
        };

        public string[] GetUnitsForCategory(string category)
        {
            return UnitsByCategory.TryGetValue(category, out var units) ? units : Array.Empty<string>();
        }
    }
}

File: java-client-resful/web-client-java-resful/src/main/java/ec/edu/grupo3/webclient/models/ConversionViewModel.java
package ec.edu.grupo3.webclient.models;

public class ConversionViewModel {
    private String category = "Mass";
    private double inputValue;
    private String fromUnit = "KILOGRAM";
    private String toUnit = "POUND";
    private Double resultValue;
    private String successMessage;
    private String errorMessage;

    public static ConversionViewModel defaultModel() {
        return new ConversionViewModel();
    }

    public String getCategory() { return category; }
    public void setCategory(String category) { this.category = category; }
    public double getInputValue() { return inputValue; }
    public void setInputValue(double inputValue) { this.inputValue = inputValue; }
    public String getFromUnit() { return fromUnit; }
    public void setFromUnit(String fromUnit) { this.fromUnit = fromUnit; }
    public String getToUnit() { return toUnit; }
    public void setToUnit(String toUnit) { this.toUnit = toUnit; }
    public Double getResultValue() { return resultValue; }
    public void setResultValue(Double resultValue) { this.resultValue = resultValue; }
    public String getSuccessMessage() { return successMessage; }
    public void setSuccessMessage(String successMessage) { this.successMessage = successMessage; }
    public String getErrorMessage() { return errorMessage; }
    public void setErrorMessage(String errorMessage) { this.errorMessage = errorMessage; }
}